### DISCUSSION TITLE
TQSL Package not found

### DISCUSSION THREAD

**Discussion by chrisdebian** [Homebrew/discussions#3455] [Linux]
Hi, all.

I feel a bit stupid posting this, but I've run out of options.

I've got a fresh install, on PopOS!, and want to install a ham radio package, called TQSL. The fresh homebrew installation completed without errors, and I completed the follow up actions that are suggested by the install script, such as amending my .profile.

I can run brew update, and brew upgrade, without issues. I've run brew doctor, without any problems, too.

When I try to install TQSL from https://formulae.brew.sh/cask/tqsl , I'm told that no cask or formula exists, with this or a similar name. I used the install command from the package page.

When I search for the package, it isn't identified, but if I substitute a different package name (Firefox), then results are returned, as expected.

I've followed the troubleshooting documentation, but have failed to work this one out. 

I had a quick look on the homebrew GitHub, to see whether the TQSL install script gave any clues, but wasn't sure how the download URL worked.

Can anyone offer any advice?

Many thanks,

Chris
chris_debian
2E0FRU

**Comment by alebcay** [comment]
Hi Chris, there's two points that you may be running into:
- In Homebrew parlance, packages can either be a _formula_ or a _cask_. A _formula_ (usually pluralized as _formulae_) is generally built/packaged from source by Homebrew CI and provided to your machine where possible, or in some situations, if your configuration does not allow for the use of prebuilt binaries that were built by Homebrew CI, it will be built locally from source on your machine. A _cask_ is a package already built by an upstream distributor which is then downloaded/unpacked on your machine by Homebrew. Casks are often GUI apps and are the distribution method of choice for closed-source/proprietary apps. **Moreover, at this time, casks are only available for macOS.**
- You can query the Homebrew website for formulae or casks, but they're two different searches. If a _formula_ for `tqsl` exists, the info would be at https://formulae.brew.sh/formula/tqsl; if a _cask_ for `tqsl` exists, the info would be at https://formulae.brew.sh/cask/tqsl. You'll notice from visiting the two links that there is no formula in the main repository named `tqsl`, but there is a cask available named `tqsl`. However, the description notes for that cask mentions that it lets you "sign and upload QSO records to Logbook of The World (LoTW)", so I'm pretty sure the `tqsl` cask bears no relation to the TQSL you're looking for.

**Reply by chrisdebian** [reply]
Hi, @alebcay .

Many thanks for your helpful answer. For some strange reason, it wasn't obvious to me that Casks were only for MacOS; my mistake.

The TQSL package that you mentioned, is the one I want, ie, for 'Logbook of the World'. Having looked at the formulae.brew page, I can see that TQSL only exists as a Cask. It would be useful if a formula was available, and I think others would benefit, so I will have a look at the documentation, and see how to create the formula.

Thanks, again.

Chris.

**Reply by carlocab** [reply]
Sorry for the confusion! It really ought to be more obvious that casks are macOS-only, and you're certainly not the only user to get tripped up by this.

**Reply by alebcay** [reply]
Thanks Chris. FWIW I took a look at building TQSL as as formula. I can manage to get the build to work but running the binaries that are created afterwards seems to result in a segfault. So I think moving forward with trying to get it to work as a formula on macOS may be a bit of an uphill struggle.

**Reply by chrisdebian** [reply]
Thanks, both. I want to use it on Linux. I'll have a go this weekend and report the outcome.

Cheers,

Chris
